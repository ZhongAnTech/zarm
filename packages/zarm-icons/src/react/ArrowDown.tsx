import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowDown = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M494.094 733.426a41.472 41.472 0 01-20.023-11.126L140.16 388.388c-16.272-16.271-16.272-42.653 0-58.925 16.271-16.272 42.653-16.272 58.925 0L503.66 634.037 808.186 329.51c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L533.2 722.347c-10.607 10.607-25.508 14.3-39.106 11.08z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowDown);
export default ForwardRef;
