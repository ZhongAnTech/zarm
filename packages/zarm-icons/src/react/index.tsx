export { default as ArrowDown } from './ArrowDown';
export { default as ArrowLeft } from './ArrowLeft';
export { default as ArrowRight } from './ArrowRight';
export { default as ArrowUp } from './ArrowUp';
export { default as CloseCircleFill } from './CloseCircleFill';
export { default as CloseCircle } from './CloseCircle';
export { default as Close } from './Close';
export { default as DeleteKey } from './DeleteKey';
export { default as Empty } from './Empty';
export { default as Keyboard } from './Keyboard';
export { default as Minus } from './Minus';
export { default as PlusCircle } from './PlusCircle';
export { default as Plus } from './Plus';
export { default as Search } from './Search';
export { default as SuccessCircle } from './SuccessCircle';
export { default as Success } from './Success';
export { default as Volume } from './Volume';
export { default as WarningCircle } from './WarningCircle';
export { default as Warning } from './Warning';
