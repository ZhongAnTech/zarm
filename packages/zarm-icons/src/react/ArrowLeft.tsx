import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowLeft = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M296.114 508.035c-3.22-13.597.473-28.499 11.079-39.105l333.912-333.912c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L395.504 498.47l304.574 304.574c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L307.241 528.058a41.472 41.472 0 01-11.127-20.023z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowLeft);
export default ForwardRef;
