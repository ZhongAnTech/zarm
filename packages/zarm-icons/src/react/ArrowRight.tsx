import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowRight = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M711.156 508.035a41.472 41.472 0 01-11.126 20.023L366.12 861.97c-16.272 16.272-42.654 16.272-58.926 0s-16.272-42.654 0-58.926L611.767 498.47 307.241 193.944c-16.272-16.272-16.272-42.654 0-58.926 16.271-16.272 42.653-16.272 58.925 0L700.078 468.93c10.606 10.606 14.299 25.508 11.078 39.105z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowRight);
export default ForwardRef;
