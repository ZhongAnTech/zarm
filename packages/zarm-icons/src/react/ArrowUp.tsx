import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowUp = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M513.176 285.05a41.472 41.472 0 0120.024 11.127L867.11 630.089c16.272 16.271 16.272 42.653 0 58.925s-42.654 16.272-58.925 0L503.612 384.44 199.085 688.967c-16.272 16.271-42.654 16.271-58.925 0-16.272-16.272-16.272-42.654 0-58.926L474.07 296.129c10.606-10.606 25.508-14.299 39.105-11.078z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowUp);
export default ForwardRef;
