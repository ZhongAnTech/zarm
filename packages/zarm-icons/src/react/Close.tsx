import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgClose = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M560.149 501.223l255.344 255.344c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L501.223 560.149 245.88 815.493c-16.272 16.272-42.654 16.272-58.925 0-16.272-16.272-16.272-42.654 0-58.926l255.344-255.344L186.954 245.88c-16.272-16.272-16.272-42.654 0-58.925 16.271-16.272 42.653-16.272 58.925 0l255.344 255.344 255.344-255.344c16.272-16.272 42.654-16.272 58.926 0 16.272 16.271 16.272 42.653 0 58.925L560.149 501.223z"
          fillRule="nonzero"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgClose);
export default ForwardRef;
