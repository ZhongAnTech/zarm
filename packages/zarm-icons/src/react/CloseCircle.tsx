import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgCloseCircle = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <g fillRule="evenodd">
          <g>
            <path d="M500 460.716l137.493-137.493c10.848-10.848 28.436-10.848 39.284 0l.101.102c10.735 10.848 10.645 28.344-.203 39.08l-138.207 136.78 138.309 138.308c10.848 10.848 10.848 28.436 0 39.284-10.848 10.848-28.436 10.848-39.284 0L498.98 538.264 359.85 675.956c-10.41 10.303-27.188 10.26-37.545-.097-10.34-10.341-10.34-27.107 0-37.448L460.716 500 323.223 362.507c-10.848-10.848-10.848-28.436 0-39.284 10.848-10.848 28.436-10.848 39.284 0L500 460.716z" />
            <path
              d="M190.64 809.36c170.855 170.854 447.865 170.854 618.72 0 170.854-170.855 170.854-447.865 0-618.72-170.855-170.854-447.865-170.854-618.72 0-170.854 170.855-170.854 447.865 0 618.72zm-44.193 44.193c-195.263-195.262-195.263-511.844 0-707.106 195.262-195.263 511.844-195.263 707.106 0 195.263 195.262 195.263 511.844 0 707.106-195.262 195.263-511.844 195.263-707.106 0z"
              fillRule="nonzero"
            />
          </g>
        </g>
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgCloseCircle);
export default ForwardRef;
