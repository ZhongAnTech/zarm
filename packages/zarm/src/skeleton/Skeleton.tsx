import * as React from 'react';
import { createBEM } from '@zarm-design/bem';
import { ConfigContext } from '../n-config-provider';
import type { HTMLProps } from '../utils/utilityTypes';
import type { SkeletonShape, SkeletonLineShape } from './interface';

// SkeletonTitle和SkeletonParagraph的shape属性值集合
const SKELETONLINESHAPE = ['radius', 'rect'];

export interface SkeletonCssVars {
  '--width'?: React.CSSProperties['width'];
  '--height'?: React.CSSProperties['height'];
  '--border-radius'?: React.CSSProperties['borderRadius'];
  '--margin'?: React.CSSProperties['margin'];
}

export type SkeletonProps = HTMLProps<SkeletonCssVars> & {
  shape?: SkeletonShape;
  animated?: boolean;
};

export type SkeletonTitleProps = HTMLProps<SkeletonCssVars> & {
  shape?: SkeletonLineShape;
  animated?: boolean;
};

export type SkeletonParagraphProps = HTMLProps & {
  shape?: SkeletonLineShape;
  animated?: boolean;
  lineCount?: number;
};

const Skeleton = (props: SkeletonProps) => {
  const { className, shape, animated, ...restProps } = props;
  const { prefixCls } = React.useContext(ConfigContext);
  const bem = createBEM('skeleton', { prefixCls });
  const cls = bem([
    {
      [`${shape}`]: !!shape,
      animated,
    },
    className,
  ]);

  return <div className={cls} {...restProps} />;
};

Skeleton.displayName = 'skeleton';

Skeleton.defaultProps = {
  shape: 'radius',
};

// SkeletonTitle
const SkeletonTitle = (props: SkeletonTitleProps) => {
  const { className, shape, ...restProps } = props;
  const { prefixCls } = React.useContext(ConfigContext);
  const bem = createBEM('skeleton', { prefixCls });
  const cls = bem('title', [className]);
  const _shape = shape && SKELETONLINESHAPE.includes(shape) ? shape : undefined;

  return <Skeleton shape={_shape} className={cls} {...restProps} />;
};

// SkeletonParagraph
const SkeletonParagraph = (props: SkeletonParagraphProps) => {
  const { className, shape, lineCount = 3, animated, ...restProps } = props;
  const { prefixCls } = React.useContext(ConfigContext);
  const bem = createBEM('skeleton', { prefixCls });
  const cls = bem('paragraph', [className]);
  const lineCls = bem('line');
  const _shape = shape && SKELETONLINESHAPE.includes(shape) ? shape : undefined;

  return (
    <div className={cls} {...restProps}>
      {lineCount > 0
        ? Array(lineCount)
            .fill(1)
            .map((...arg) => (
              <Skeleton key={arg[1]} shape={_shape} animated={animated} className={lineCls} />
            ))
        : null}
    </div>
  );
};

export default Skeleton;
export { SkeletonTitle, SkeletonParagraph };
