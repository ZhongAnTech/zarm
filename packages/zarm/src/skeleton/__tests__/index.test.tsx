import * as React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import Skeleton from '../index';

describe('Skeleton', () => {
  afterEach(cleanup);

  test('should render with default props', () => {
    const { container } = render(<Skeleton />);
    expect(container).toMatchSnapshot();
  });

  it('should render radius shape skeleton', () => {
    render(<Skeleton data-testid="skeleton-root" />);
    expect(screen.getByTestId('skeleton-root')).toBeTruthy();
    expect(screen.getByTestId('skeleton-root').className).toEqual(
      'za-skeleton za-skeleton--radius',
    );
  });

  test.each`
    className            | shape       | expected
    ${'custom-skeleton'} | ${'circle'} | ${'za-skeleton za-skeleton--circle custom-skeleton'}
    ${'custom-skeleton'} | ${'radius'} | ${'za-skeleton za-skeleton--radius custom-skeleton'}
    ${'custom-skeleton'} | ${'rect'}   | ${'za-skeleton za-skeleton--rect custom-skeleton'}
  `(
    'should render with className - $className and shape - $shape',
    ({ className, shape, expected }) => {
      render(<Skeleton data-testid="skeleton-root" className={className} shape={shape} />);
      expect(screen.getByTestId('skeleton-root').className).toEqual(expected);
    },
  );

  it('should render animated skeleton', () => {
    render(<Skeleton animated data-testid="skeleton-root" />);
    expect(screen.getByTestId('skeleton-root')).toBeTruthy();
    expect(screen.getByTestId('skeleton-root').className).toEqual(
      'za-skeleton za-skeleton--radius za-skeleton--animated',
    );
  });

  it('should render Skeleton.Title', () => {
    render(<Skeleton.Title data-testid="skeleton-root" />);
    expect(screen.getByTestId('skeleton-root')).toBeTruthy();
    expect(screen.getByTestId('skeleton-root').className).toEqual(
      'za-skeleton za-skeleton--radius za-skeleton__title',
    );
  });

  it('should render rect Skeleton.Title', () => {
    render(<Skeleton.Title shape="rect" data-testid="skeleton-root" />);
    expect(screen.getByTestId('skeleton-root')).toBeTruthy();
    expect(screen.getByTestId('skeleton-root').className).toEqual(
      'za-skeleton za-skeleton--rect za-skeleton__title',
    );
  });

  it('should render animated Skeleton.Title', () => {
    render(<Skeleton.Title animated data-testid="skeleton-root" />);
    expect(screen.getByTestId('skeleton-root')).toBeTruthy();
    expect(screen.getByTestId('skeleton-root').className).toEqual(
      'za-skeleton za-skeleton--radius za-skeleton--animated za-skeleton__title',
    );
  });

  it('should render Skeleton.Paragraph', () => {
    render(<Skeleton.Paragraph lineCount={5} data-testid="skeleton-root" />);
    expect(screen.getByTestId('skeleton-root')).toBeTruthy();
    expect(screen.getByTestId('skeleton-root').className).toEqual('za-skeleton__paragraph');
    expect(screen.getByTestId('skeleton-root').firstElementChild!.className).toEqual(
      'za-skeleton za-skeleton--radius za-skeleton__line',
    );
    expect(screen.getByTestId('skeleton-root').childElementCount).toBe(5);
  });

  it('should render animated Skeleton.Paragraph', () => {
    render(<Skeleton.Paragraph animated data-testid="skeleton-root" />);
    expect(screen.getByTestId('skeleton-root')).toBeTruthy();
    expect(screen.getByTestId('skeleton-root').firstElementChild!.className).toEqual(
      'za-skeleton za-skeleton--radius za-skeleton--animated za-skeleton__line',
    );
  });

  it('should render rect Skeleton.Paragraph', () => {
    render(<Skeleton.Paragraph shape="rect" data-testid="skeleton-root" />);
    expect(screen.getByTestId('skeleton-root')).toBeTruthy();
    expect(screen.getByTestId('skeleton-root').firstElementChild!.className).toEqual(
      'za-skeleton za-skeleton--rect za-skeleton__line',
    );
  });
});
