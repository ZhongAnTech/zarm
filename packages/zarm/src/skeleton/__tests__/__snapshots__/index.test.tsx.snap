// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should render with default props 1`] = `
<div>
  <div
    class="za-skeleton za-skeleton--radius"
  />
</div>
`;
